.github/workflows/validate-pr.yml

```diff
@@ -109,7 +109,7 @@ jobs:
     steps:
       - uses: actions/setup-python@v5
         with:
-          python-version: 3.11
+          python-version: 3.12
```

Comment:
Shouldn't we rather rely on a python matrix there too?

Reply:
You mean, running the benchmark with all versions of Python? 
I guess it wasn't done for cost effectiveness reasons, but it can be changed, indeed.

Reply:
Yes, that's what I meant! (cf: comment on benchmark requirements)